app/src/main/cpp/skyline/jvm.h

```diff
@@ -106,6 +108,26 @@ namespace skyline {
          */
         void ClearVibrationDevice(jint index);
 
+        /**
+         * @brief A call to EmulationActivity.showKeyboard in Kotlin
+         */
+        jobject ShowKeyboard(applet::swkbd::KeyboardConfigVB& config, std::u16string initial_text);
+
+        /**
+         * @brief A call to EmulationActivity.waitForSubmitOrCancel in Kotlin
+         */
+        std::pair<applet::swkbd::CloseResult,std::u16string> WaitForSubmitOrCancel(jobject dialog);
```

Comment:
Needs space, not sure about using swkbd types in jvmmanager tbh @PixelyIon thoughts?

Reply:
I'm not necessarily revolted by it and think it's passable, the most optimal thing to do would be to duplicate the enum here and write a switch-case function to convert which would end up just being passthrough. I'm fine with retaining it as-is though as ultimately it is still somewhat redundant and it's not like the frontend isn't already dealing with implementation details of the guest applet.